src/test/ui/consts/const_in_pattern/cross-crate-pass.rs

```diff
@@ -0,0 +1,14 @@
+// run-pass
+// aux-build:consts.rs
+
+#![warn(indirect_structural_match)]
+
+extern crate consts;
+use consts::*;
+
+fn main() {
+    match Some(CustomEq) {
+        NONE => panic!(),
```

Comment:
maybe write the pattern as `consts::NONE` to make it crystal clear to future readers that this is not a binding to a local ident?

(For some reason I have a preference for that path-y style when using a cross-crate const in a pattern...)